filters/filters.go

```diff
@@ -0,0 +1,94 @@
+package filters
+
+import (
+	"net/http"
+
+	"github.com/getlantern/errors"
+
+	"github.com/getlantern/http-proxy/utils"
+)
+
+// Filter is like an http.Handler that
+type Filter interface {
+	// Apply is like the function on http.Handler but also gets a Next which
+	// allows it to continue execution along the current filter chain. If Apply
+	// returns an error, we will write an appropriate status code to the response.
+	// Tip - use filters.Fail() to provide an error with a description.
+	Apply(w http.ResponseWriter, req *http.Request, next Next) error
```

Comment:
Yes, I like the API. And thanks for considering the suggestion of naming for "context" to "next"!